pkg/orchestration/k8s/nodegroup.go

```diff
@@ -69,6 +70,24 @@ func (g *NodeGroup) AddNode(name string, o orchestration.NodeOptions) (err error
 		Restricted:          config.Restricted,
 	})
 
+	// pregenerate Swarm key if needed
+	if o.SwarmKey == "" && config.Transaction == "" && (!config.SwapEnable || config.ChequebookEnable) {
+		swarmKey, overlay, err := utils.CreateSwarmKey(config.Password)
+		if err != nil {
+			return fmt.Errorf("create Swarm key for node %s: %w", name, err)
+		}
+
+		txHash, err := g.cluster.swap.AttestOverlayEthAddress(ctx, overlay)
```

Comment:
make sure there is a log line for this result. either here or inside the geth service.

Reply:
Done.